File: search_algorithms/linear_search.py
# Implementation of Linear Search Algorithm

def linear_search(arr, target):
    n = len(arr)
    for i in range(n):
        if arr[i] == target:
            return i
    return -1


if __name__ == "__main__":
    arr = [23, 2, 5, 99, 340, 734, 200, 456, 789, 123, 456, 789, 1000]
    target = 123
    result = linear_search(arr, target)
    print(f"Element {target} found at index {result}")

File: search_algorithms/linear_search.c
#include <stdio.h>


int linear_search(int arr[], int n, int target)
{
    for (int i = 0; i < n; i++)
    {
        if (arr[i] == target)
        {
            return i;
        }
    }
    return -1;
}

int main()
{
    int arr[] = {21, 93, 18, 23, 99, 112, 666};
    int n = sizeof(arr) / sizeof(arr[0]);

    int target = 112;
    printf("Element %d found at index %d\n", target, linear_search(arr, n, target));

}